.github/workflows/tests.yaml

```diff
@@ -0,0 +1,38 @@
+name: test
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "**"
+    paths-ignore:
+      - "docs/**"
+
+concurrency:
+  group: build-test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "Unit and Integration Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install system packages
+        run: sudo apt-get install -y portaudio19-dev
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install -r requirements-full.txt
```

Comment:
is there a way to extract the requirements from the toml file for testing? Otherwise we'll need to keep it in sync with `requirements-full.txt` which seems like it might be error-prone.

Reply:
Aha I just tried and it does run the tests, but it seems to miss the doctests (7 tests with pytest vs 11 with unittest)

Reply:
Oh, I see.  Let me see what I can do...